### PR TITLE
Pull Request : Deep Learning CNN Model for Tomato Leaf Disease Predic…

### LINKED ISSUE #260: Deep Learning CNN Model for Tomato Leaf Disease Prediction
# 🌱 New Project: Deep Learning CNN Model for Tomato Leaf Disease Prediction

## 📌 Description

This project introduces a **Deep Learning-based Convolutional Neural Network (CNN)** model trained on a **Tomato Leaf Image Dataset** to detect and classify various tomato plant diseases. The model can distinguish between multiple disease categories as well as healthy leaves, helping farmers and researchers in early disease detection and crop protection.

---

## 🦠 Diseases Covered

The trained CNN model can accurately classify tomato leaves into the following categories:

* Bacterial spot
* Early blight
* Late blight
* Leaf mold
* Septoria leaf spot
* Spider mites (Two-spotted spider mite)
* Target spot
* Tomato Yellow Leaf Curl Virus
* **Healthy Tomato Leaf** 🌿

---

## 📊 Model Performance

* ✅ Training Accuracy: **99.96%**
* ✅ Testing Accuracy: **96.99%**
* ⚖️ The model is **not overfit or underfit**, indicating strong generalization.

---

## 🖼️ Sample Outputs

* Healthy Tomato Leaf

Image

* Tomato Leaf with Bacterial Spot

Image

* Tomato Leaf with Septoria Leaf Spot

Image

* Tomato Leaf with Tomato Yellow Leaf Curl Virus

Image

---

## 📂 Repo Placement

```
projects/Day-XXX-Tomato-Disease-Prediction-CNN/
│── dataset/
│── model/
│── tomato_cnn.py
│── README.md
```

---

## ⚙️ Tech Stack

* **Python 3**
* **TensorFlow / Keras** (Deep Learning)
* **OpenCV / Pillow** (Image Preprocessing)
* **NumPy, Pandas** (Data Handling)
* **Matplotlib / Seaborn** (Visualization)

---

## ▶️ How to Run

1. Clone the repository
2. Navigate to the project folder
3. Install dependencies:

   ```bash
   pip install tensorflow keras numpy pandas matplotlib seaborn opencv-python
   ```
4. Train or load the pretrained model:

   ```bash
   python tomato_cnn.py
   ```
5. Test on sample images

---

## 🚀 Future Enhancements

* Deploy as a **Streamlit web app** for user-friendly predictions
* Add **real-time detection** using camera feed
* Extend the dataset with more crop varieties
* Provide **disease treatment recommendations** along with predictions

---

🔗 **Event:** GirlScript Summer of Code (GSSoC’25)
🏷️ **Type:** New Project Addition

---

### PR DESCRIPTION
**Pull request by keshripritesh**
# 🌱 Add Tomato Leaf Disease Prediction using CNN

## 📌 Related Issue

Closes #260 

---

## 📖 Project Description

This PR adds a new project **Deep Learning CNN Model for Tomato Leaf Disease Prediction** under **GirlScript Summer of Code (GSSoC’25)**.
The model is trained on a **Tomato Leaf Image Dataset** and can predict multiple tomato plant diseases as well as identify healthy leaves.

---

## ✨ Features Implemented

* 🦠 Predicts **8 tomato diseases + healthy leaf**
* 📊 Achieves **99.96% training accuracy** and **96.99% testing accuracy**
* 🔄 Balanced training (model is **not overfit/underfit**)
* 🖼️ Supports image preprocessing (`50x50x3`)
* 📂 Structured dataset and model pipeline included

---

## 🦠 Diseases Covered

* Bacterial Spot
* Early Blight
* Late Blight
* Leaf Mold
* Septoria Leaf Spot
* Spider Mites (Two-Spotted Spider Mite)
* Target Spot
* Tomato Yellow Leaf Curl Virus
* Healthy Tomato Leaf 🌿

---

## ▶️ How to Run

1. Clone the repository
2. Navigate to the project folder
3. Install dependencies:

   ```bash
   pip install tensorflow keras numpy pandas matplotlib seaborn opencv-python
   ```
4. Train or load the pretrained model:

   ```bash
   python tomato_cnn.py
   ```
5. Test predictions on sample images

---

### ✅ Model Accuracy
Model Accuracy

### 🏗️ Model Architecture
Model Architecture

---

## 🚀 Future Enhancements

* Build a **Streamlit web app** for interactive predictions
* Integrate **real-time camera-based detection**
* Expand dataset for more crop disease detection
* Add **treatment recommendation system**

---

## ✅ Checklist

* [x] Added CNN model with training + testing pipeline
* [x] Included dataset and preprocessing steps
* [x] Output snapshots added (accuracy & architecture)
* [x] README updated with instructions & disease details

---

👩‍💻 **Contributor:** @keshripritesh (Pritesh Keshri)
🔗 **Event:** GirlScript Summer of Code (GSSoC’25)

---

### CODE REVIEW

**Comment by keshripritesh** [comment]
> Send your name email id and github id

Name : Pritesh Keshri
Email id : priteshkeshri468@gmail.com
Github : keshripritesh (https://github.com/keshripritesh)
